### ISSUE TITLE
World.collidePoint (and probably others) ignore Entity.collidable

### ISSUE THREAD

**Issue by Draknek** [useflashpunk/FlashPunk#66]
This is super-embarrassing!  Should be fixed ASAP.

**Comment by zachwlewis** [comment]
Is this issue fixed?

``` as3
/**
 * Returns the first Entity found that collides with the position.
 * @param   type        The Entity type to check for.
 * @param   pX          X position.
 * @param   pY          Y position.
 * @return  The collided Entity, or null if none collide.
 */
public function collidePoint(type:String, pX:Number, pY:Number):Entity
{
    var e:Entity = _typeFirst[type];
    while (e)
    {
        if (e.collidable && e.collidePoint(e.x, e.y, pX, pY)) return e;
        e = e._typeNext;
    }
    return null;
}
```

It seems like all of the collision calls from `Entity` all ignore the calling entity's collision. This is useful in some cases (such as how `Entity.onCamera()` relies on `Entity.collideRect()`), but it goes against reason to return anything other than `null` when being called from an entity with collision set to false.

**Comment by Draknek** [comment]
It was fixed for that method but still broken elsewhere! Oops, so much for "ASAP"

Fixed in 5c090f066235af83f8fede300ac544419f27283d and db37844925496824164d07b2f7015810940647de

**Comment by zachwlewis** [comment]
ASAP just so happened to be "in about a year," haha.
—
